### PR TITLE
Support present hash

### PR DESCRIPTION
**Pull request by u2**
#955

### CODE REVIEW

File: lib/grape/dsl/inside_route.rb [hunk 1]
```diff
@@ -184,29 +184,37 @@ def body(value = nil)
...
+
+          representation = { root => representation } if root
+        end
 
-        representation = { root => representation } if root
         if key
           representation = (@body || {}).merge(key => representation)
         elsif entity_class.present? && representation.respond_to?('merge')
           representation = (@body || {}).merge(representation)
+        elsif hash_present && representation.respond_to?('merge')
```
Comment (dblock) [inline lib/grape/dsl/inside_route.rb]: Should this be `respond_to?(:merge)`?

**Comment by dblock** [comment]
So, the real question given this code is why isn't `present <hash>` a noop to return the hash? What are we gaining by running it through `present`?

**Comment by dblock** [comment]
Another question is what does `present <hash>, with: Presenter` do and does it make any sense?

**Comment by u2** [comment]
Yes, `present <hash>` just merge `body` and return the `body`, no other thing.

File: lib/grape/dsl/inside_route.rb (hunk 1 above)
Comment (u2) [inline lib/grape/dsl/inside_route.rb]: No, removed already.

**Comment by u2** [comment]
Yes, maybe it's better to `present <hash>, with: Presenter`.I will try to rewrite it.

**Comment by dblock** [comment]
I think I am missing something. What is the difference in the result between `present <hash>` and `present <hash>, with: Presenter`?

**Comment by u2** [comment]
For the result, there is no difference.But `present <hash>, with: Presenter` fit the grape style.Just add a `Presenter`, and use `with: Presenter` for represent the object itself.It can present anything, not only `Hash`.

**Comment by dblock** [comment]
Hm, I can see the argument for a noop class like this. Do you think it belongs in `Middleware`? It's not a Middleware.

(and sorry to be difficult :)

**Comment by u2** [comment]
I was hesitant to about that, but I am not sure where I can place `Presenter `.

**Comment by dblock** [comment]
Maybe just `Grape::Presenters::Presenter`?

**Comment by u2** [comment]
Ok, I will change it right now. :)

File: spec/grape/dsl/inside_route_spec.rb [hunk 2]
```diff
@@ -268,6 +268,46 @@ def initialize
...
+            end
+
+            describe 'instance' do
+              before do
+                subject.present hash_mock1, with: Grape::Presenters::Presenter
+                subject.present hash_mock2, with: Grape::Presenters::Presenter
+              end
+
+              it 'presents both dummy presenter' do
+                puts subject.body.inspect
```
Comment (dblock) [inline spec/grape/dsl/inside_route_spec.rb]: Stray `puts` here. 

Lets put this into a separate _spec file too.

**Comment by dblock** [comment]
I think the code here looks perfect. This needs an entry in README. Maybe just describe something around using this presenter to keep things consistent?

**Comment by u2** [comment]
I have updated README.

**Comment by dblock** [comment]
Merging, thanks for your patience.

**Comment by swistaczek** [comment]
Hey, I still dont get idea behind this change :dancer:

**Comment by dblock** [comment]
It's entirely to make returning data consistent, if you `present` an object you should also be able to `present` anything else. Sometimes it makes sense if your code path figures out what to present, then presents it.

**Pull request merged**